web/packages/teleterm/src/ui/services/connectionTracker/connectionTrackerService.ts

```diff
@@ -162,7 +163,20 @@ export class ConnectionTrackerService extends ImmutableStore<ConnectionTrackerSt
     });
   }
 
-  async disconnectAndRemove(connections: TrackedConnection[]): Promise<void> {
+  async disconnectAndRemoveItemsBelongingToResource(
+    resourceUri: uri.ResourceUri
+  ): Promise<void> {
+    const connections = this.getConnections().filter(s => {
+      if (routing.parseServerUri(resourceUri)) {
+        return s.kind === 'connection.server' && s.serverUri === resourceUri;
+      }
+      if (
+        routing.parseDbUri(resourceUri) ||
+        routing.parseKubeUri(resourceUri)
+      ) {
+        return s.kind === 'connection.gateway' && s.targetUri === resourceUri;
+      }
```

Comment:
Wouldn't it be better to switch based on `connection.kind` and look at `serverUri` / `targetUri`? This way the compiler will automatically tell you to expand this if we ever add a new connection kind.

Checking the kind of the resource URI wouldn't be necessary in this situation, I think.

Reply:
Yeah, much better. I don't know why I chose so convoluted way to do it.